### PR TITLE
PERF: TimedeltaArray.__iter__

### LINKED ISSUE #26713: Performance for TimedeltaArray.__iter__ and PeriodArray.__iter__
Right now `DatetimeArray.__iter__` does it's boxing in chunks. We could do something similar for TimedeltaArray and possibly PeriodArray.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #26713
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Also 2d compat for TDA/DTA `__iter__`

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -558,19 +558,21 @@ def __iter__(self):
...
-        chunksize = 10000
-        chunks = int(length / chunksize) + 1
-        for i in range(chunks):
-            start_i = i * chunksize
-            end_i = min((i + 1) * chunksize, length)
-            converted = ints_to_pydatetime(
-                data[start_i:end_i], tz=self.tz, freq=self.freq, box="timestamp"
-            )
-            yield from converted
+        if self.ndim > 1:
```
Comment (jreback) [inline pandas/core/arrays/datetimes.py:561]: umm, when is this hit?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py:561]: it isn't ATM, but there do exist 2D DTAs so we should handle this correctly

Comment (jreback) [reply pandas/core/arrays/datetimes.py:561]: ok, either leave it out or pls add a test (former is prob better); can followon later i think.

Comment (jbrockmendel) [reply pandas/core/arrays/datetimes.py:561]: this has a test test_iter_2d

**Comment by jbrockmendel** [comment]
gentle ping

**Review by jreback — APPROVED** [review]

**Pull request merged**